### PR TITLE
feature (ref 35952): use different method for statement with or witho…

### PR DESCRIPTION
**Pull request by jalalifardemos**
**Ticket:** https://yaits.demos-deutschland.de/T35952

Description: Use different Method to create Statement with or without files attachment. 

### How to review/test
In Browser or Code review

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php [hunk 1]
```diff
@@ -3019,8 +3019,11 @@ private function attachSimilarStatementSubmitters(Statement $statementToAttachTo
     public function createNonOriginalStatement(array $originalStatementData, Statement $newOriginalStatement): Statement
     {
         $fieldsForUpdateStatement = $this->extractFieldsForUpdateStatement($originalStatementData);
-        $copyOfStatement = $this->statementCopier->copyStatementObjectWithinProcedureWithRelatedFiles($newOriginalStatement, false, true);
-
+        if ($newOriginalStatement->getFiles()) {
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php:3022]: Wouldn't it be better to have this logic within `copyStatementObjectWithinProcedureWithRelatedFiles`? It should be possible to check there whether files are set or not

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php (hunk 1 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php]: done in [1d8d5cb](https://github.com/demos-europe/demosplan-core/pull/2515/commits/1d8d5cb490dd5c96aa187f962b7a6e7b7edc2ae1)

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php [hunk 2]
```diff
@@ -508,25 +507,27 @@ public function copyStatementObjectWithinProcedureWithRelatedFiles(
...
+            // persist to get an ID for the FileContainer copying below
+            $this->getDoctrine()->getManager()->persist($newStatement);
 
-        $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
+            $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
 
-        return $newStatement;
+            return $newStatement;
+        } else {
+            return $this->copyStatementObjectWithinProcedure($statement);
```
Comment (mussbach) [inline demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php:529]: Why didn't you pass `$createReport` and `$copyOnCreateStatement`?

**Review by mussbach — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php (hunk 2 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php]: I thought, they have already defaulted values. 
done in ([404d7ec](https://github.com/demos-europe/demosplan-core/pull/2515/commits/404d7ec2692b2af591f0bbf981fa7be9c62426d3))

**Review by mussbach — APPROVED** [review]
lgtm, thx, code review only

**Pull request merged**